reconcile/database_access_manager.py

```diff
@@ -0,0 +1,621 @@
+    engine: str,
+    settings: dict[Any, Any],
+) -> None:
+    if state.exists(db_access.name):
+        current_state = state.get(db_access.name)
+        if current_state == db_access.dict(by_alias=True):
+            return
+
+    resource_prefix = f"dbam-{db_access.name}"
+    oc = oc_map.get_cluster(cluster_name, False)
+
+    database_connection = _create_database_connection_parameter(
+        db_access,
+        namespace_name,
+        oc,
+        admin_secret_name,
+        resource_prefix,
+    )
+
+    managed_resources = _populate_resources(
+        db_access,
+        engine,
+        settings["imageRepository"],
+        settings["pullSecret"],
+        admin_secret_name,
+        resource_prefix,
+        settings,
+        database_connection,
+    )
+
+    # create job, delete old, failed job first
+    job = oc.get(
+        namespace_name,
+        "Job",
+        f"dbam-{db_access.name}",
+        allow_not_found=True,
+    )
+    if not job:
+        for r in managed_resources:
+            openshift_base.apply(
+                dry_run=dry_run,
+                oc_map=oc_map,
+                cluster=cluster_name,
+                namespace=namespace_name,
+                resource_type=r.resource.kind,
+                resource=r.resource,
+                wait_for_namespace=False,
+            )
+        return
+    job_status = JobStatus(
+        conditions=[
+            JobStatusCondition(type=c["type"])
+            for c in job["status"].get("conditions", [])
+        ]
+    )
+    if job_status.is_complete():
+        if job_status.has_errors():
+            raise JobFailedError(f"Job dbam-{db_access.name} failed, please check logs")
+        state.add(
+            db_access.name,
+            value=db_access.dict(by_alias=True),
+            force=True,
+        )
+        logging.debug("job completed, cleaning up")
+        for r in managed_resources:
+            if r.clean_up:
+                openshift_base.delete(
+                    dry_run=dry_run,
+                    oc_map=oc_map,
+                    cluster=cluster_name,
+                    namespace=namespace_name,
+                    resource_type=r.resource.kind,
+                    name=r.resource.name,
+                    enable_deletion=True,
+                )
+    else:
+        logging.info(f"Job dbam-{db_access.name} appears to be still running")
+
+
+class DatabaseAccessManagerIntegration(QontractReconcileIntegration):
+    @property
+    def name(self) -> str:
+        return QONTRACT_INTEGRATION
+
+    def run(self, dry_run: bool) -> None:
+        settings = queries.get_app_interface_settings()
+
+        sql_query_settings = settings.get("sqlQuery")
+        if not sql_query_settings:
+            raise KeyError("sqlQuery settings are required")
+
+        state = init_state(
+            integration=QONTRACT_INTEGRATION, secret_reader=self.secret_reader
+        )
+
+        encounteredErrors = False
```

Comment:
`encounteredErrors` -> `encountered_errors`, instead of a bool, can collect all error objects then raise a `ExceptionGroup` to contain error details.